### PR TITLE
V2 : Enable coverage reports

### PR DESCRIPTION
**Pull request by z3dev**
These changes enables coverage reports for packages with test suites.

The coverage reports can be viewed by opening the "coverage_reports.html" file in the project directory.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,6 @@
 node_modules/
 components/
-converage/
+coverage/
```
Comment (kaosat-dev) [inline .gitignore]: doh ! nice catch!

File: package.json [hunk 2]
```diff
@@ -35,6 +36,7 @@
     "@jscad/sample-files": "github:jscad/sample-files#master",
     "ava": "1.4.0",
     "lerna": "3.19.0",
+    "nyc": "^15.0.0",
```
Comment (kaosat-dev) [inline package.json]: we should always use exact versions (so without the caret "^" symbol); as not exact versions of dependencies can lead to some issues (breakeage, unanonced and not correct api changes etc)

File: coverage_reports.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<!doctype html>
```
Comment (kaosat-dev) [inline coverage_reports.html]: hmm why do we need this html file ? or even the coverage_reports folder at all ? (and we even .gitignore the coverage folder ?)

Comment (z3dev) [reply coverage_reports.html]: I added the generation of HTML to the coverage reports, which allows some very nice browsing of coverage per file. However, the HTML is located in the package specific ‘coverage’ folder.

That’s why I added this index to the packages. I thought you may have questions about this. 😃 

Can you try out generating the reports, and browse a little?

File: package.json (hunk 2 above)
Comment (z3dev) [reply package.json]: done

File: coverage_reports.html (hunk 3 above)
Comment (kaosat-dev) [reply coverage_reports.html]: haha I'll take it for a spin locally to generate reports and take a look tomorow then !

**Comment by kaosat-dev** [comment]
@z3dev tried to run this today, but sadly, as I feared , it does not work unless you add nyc as a dependency of all the sub packages (I think it worked for you because you had a version with nyc already installed in there, you will likely have the same issue if you run **lerna clean** before running the coverage command).

I understand why you wanted to remove the dependency from each individual package, but I fear it is not that trivial (I tried to do this a while back with ava , which is also a similar devdependency) and failed

**Comment by kaosat-dev** [comment]
there is a lerna command to achieve something like that, trying to solve it now 
https://github.com/lerna/lerna/blob/master/doc/hoist.md

**Comment by kaosat-dev** [comment]
aha ! Found a solution ! Shall I edit your PR a tiny bit @z3dev  ? it is just a minor change in the main package.json

**Comment by kaosat-dev** [comment]
@z3dev  ?

**Comment by z3dev** [comment]
> aha ! Found a solution ! Shall I edit your PR a tiny bit @z3dev ?

@kaosat-dev Please amend.

**Comment by z3dev** [comment]
@kaosat-dev looks good.

I thought you might use hoist. Do you know if lerna.json can be used for hoist as well?

**Comment by kaosat-dev** [comment]
@z3dev , hmm good question, the docs are not clear on the subject at all : does it matter if it is in the command or lerna config ?

**Comment by z3dev** [comment]
Nope. I’m fine. We are both learning a lot. 😃

**Comment by z3dev** [comment]
@kaosat-dev please merge.

**Pull request merged**